### PR TITLE
Fixes Empty Label Identifer Preview in Settings/DataModel/Object/Edit

### LINKED ISSUE #6143: Identifier Preview is Empty in Object Settings
# Current Behavior

In the Object settings, the "Identifier preview" is empty.

CleanShot 2024-07-05 at 18 33 58@2x

# Desired Behavior

We should display the first record of the table as a sample in the identifier preview.

If there is no record yet, we should display placeholder content such as "Text" for a text field or "12345" for a number field.

CleanShot 2024-07-05 at 18 37 03@2x

### PR DESCRIPTION
**Pull request by Faisal-imtiyaz123**
fixes #6143

### CODE REVIEW

File: packages/twenty-front/src/pages/settings/data-model/SettingsObjectEdit.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (FelixMalfait) [inline packages/twenty-front/src/pages/settings/data-model/SettingsObjectEdit.tsx:1]: We don't want to disable rules like this

**Comment by FelixMalfait** [comment]
Hey @Faisal-imtiyaz123, I tried to review this but I couldn't figure out exactly what you did and why you did it.
We really value simple and elegant changes in the code base. Do you think you could do this fix without changing as many things, or explain why you went through this path in the PR description? Otherwise it's a bit hard to follow

**Comment by FelixMalfait** [comment]
Because you don't follow conventions very well (like naming the hook "[previewRecordIdHooks.ts]") then it makes the reviewer suspicious about the rest of the code, which might indeed be the best solution possible, but in that case it's best to provide context and explain why you had to make those choices

**Comment by FelixMalfait** [comment]
I will close the PR for now to keep the PR backlog clean but please feel free to raise a new one with a clean description and the most elegant solution possible without introducing un-necessary hooks or complex conditions. Thanks!

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait There were couple of problems with this one. One was that `FormProvider` was not wrapped with `RecordFieldValueSelectorContextProvider` which was showing nothing at all in Preview. The other issue was with the `form label ` selected. If u changed default `form label` let's say `name` to `form label` `custom` and did a refresh or navigated back , then when u come to this page again `custom` will appear as  selected , but it is not because `select` component had  no `default value`

**Comment by Faisal-imtiyaz123** [comment]
The hooks I made to use the previewRecordId's. Not much code there . Just for refactoring I made them.

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait I have accidentally pushed someone else's commits also. Very sorry for this.

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait If I raise I new PR I think  I should only change `previewRecordIdHooks` file, by removing it and replacing it with two files `usePreviewRecordId.ts` and `useSetPreviewRecordId.ts` and create a folder `state` under `preview` folder to store the `previewRecordIdState`

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait I would request you to please give it another look according to what I explained. I put quite some work in it.

**Comment by FelixMalfait** [comment]
@Faisal-imtiyaz123

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait Thanks for reopening it. Do u want me to make a new PR with the refactoring of `PreviewRecordIdHooks` as I explained in this comment --->>>
> @FelixMalfait If I raise I new PR I think I should only change `previewRecordIdHooks` file, by removing it and replacing it with two files `usePreviewRecordId.ts` and `useSetPreviewRecordId.ts` and create a folder `state` under `preview` folder to store the `previewRecordIdState`

Or Are you currently looking into it ? Please let me know.

**Comment by FelixMalfait** [comment]
Sorry I should not have closed it. I think you are a great problem solver, but since your code is sometimes not the cleanest it's easy to judge quickly and dismiss it. I should have spent more time on it. We'll merge it like this!

**Comment by FelixMalfait** [comment]
@lucasbordeau when you come back I'd appreciate if you can have a look at this merged PR

**Review by FelixMalfait — APPROVED** [review]

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait I am deeply thankful to you for reopening it again and spending time to  review it . Thanks for believing in me.
